src/50_Samples_%26_Templates/Product.html

```diff
@@ -335,69 +335,71 @@ <h2 class="m1">Apple</h2>
     corresponding to the selected colour. We realized the color selection by using `amp-selector`. Find more the doc and info in The Product Page section below.
 
     The green colour is the default color for the gallery when the page loads.
-    Clicking on a preview image will reveal the image in the gallery via the tap action `on="tap:product-carousel-preview.goToSlide(index=0)"`.
+    Clicking on a preview image will reveal the image in the gallery via the tap action `on="tap:AMP.setState({selectedSlide: 0})"`.
 
     See below for more informations on how we use`amp-bind`. -->
   <div class="product-gallery">
-    <ul class="show" [class]="(!selectedColor || selectedColor == 'green') ? 'show' : 'hide'">
+    <ul class="show" [class]="(!product.selectedColor || product.selectedColor == 'green') ? 'show' : 'hide'">
```

Comment:
Bind expressions don't evaluate until the user takes some action to change bind's state. 

With that particular snippet, I think you want 

```
<p class="price-description">Price: <span [text]="product[product.selectedColor].sizes[product.selectedSize]">$1.99</span> </p>
```

Since both of those vars live on the new `amp-state` element.